server/service/src/report/report_service.rs

```diff
@@ -167,6 +172,70 @@ fn generate_html_report_to_html(
     Ok(file.id)
 }
 
+/// Converts the report to an Excel file and returns the file id
+fn print_html_report_to_excel(
+    base_dir: &Option<String>,
+    document: GeneratedReport,
+    report_name: String,
+) -> Result<String, ReportError> {
+    let sheet_name = "Report";
+
+    let mut book = umya_spreadsheet::new_file();
+    let _ = book
+        .set_sheet_name(0, sheet_name)
+        .map_err(|err| ReportError::DocGenerationError(format!("{}", err)))?;
+
+    let fragment = Html::parse_fragment(&format_html_document(document));
+    let container_selector = Selector::parse(r#"div[class="container"]"#).unwrap();
+    let table_head_selector = Selector::parse("thead").unwrap();
+    let table_body_selector = Selector::parse("tbody").unwrap();
+    let row_selector = Selector::parse("tr").unwrap();
+    let cell_selector = Selector::parse("td").unwrap();
```

Comment:
I wanted to add some suggestions and check out the functionality in more detail. I realise you have to rebuild are re-run server every time any changes happen here, at the start I wanted to show you how to add quick tests and check functionality via tests rather then by re-running the whole thing. I think it's good to have this also for future devs to quickly see how things work and try changes themselves. This is kind of what I mean (at least for selectors):

https://github.com/msupply-foundation/open-msupply/compare/3098-excel-from-html...3098-excel-from-html(Test-example)

Reply:
You'll also notice in that branch, that i do `let cell` rather then `get_cell_mut` twice and also `let sheet` rather then repeated `get_sheet_by_name_mut`